### ISSUE TITLE
Blazor Desktop Components Not Supporting Partial Classes for defining code in the code behind partial class

### ISSUE THREAD

**Issue by musictopia2** [dotnet/aspnetcore#31715]
When i create a partial class and define a method in that partial class, the code won't even compile.

Good news is it works fine if using a razor class library.  The bad news is in blazor desktop does not work currently.
I hope the problem can be fixed.  Otherwise, it would need to be clear that developers would have to know that if they want to split out into partial classes and have code in the partial class then they have to use a razor class library.

**Comment by musictopia2** [comment]
I have a little more information to provide about this problem.
I found out that if the project has any xaml files, then when i try to compile, then it will give compile errors.  For example, if I override the OnInitialized on the component and compile, it will say it does not recognize the method.  Even if I have custom methods like OnClick, same thing.  However, if I take out the xaml file, then its okay.  However, that means that a person for now can only do code behind if using a razor class library.

**Comment by musictopia2** [comment]
I would like to know if you want me to create a simple repository to show that in every case, if you have a xaml file in a class library then partial classes with code behind does not work compile (even overriding the OnInitialized) in the partial class.

**Comment by Eilon** [comment]
@musictopia2 thank you for the bug report. This is specifically in WPF projects? If so, it's being tracked by this issue: https://github.com/dotnet/wpf/issues/4421. That issue also describes an easy workaround that can unblock you in the meantime.

**Comment by musictopia2** [comment]
I have a question.  Can WPF be used for blazor desktop when .net 6 is released?  I hope so because there is one scenario that requires wpf that winui does not have and that is the ability to play mp3, videos.  Or even being able to open popup windows.  What I ended up doing for my workaround is creating a razor class library and having the project just reference that one.  When I did that, then I was able to use partial classes for no problem.  That was an acceptable workaround.
